pss/pss.go

```diff
@@ -219,21 +287,29 @@ func (p *Pss) Start(srv *p2p.Server) error {
 			}
 		}
 	}()
-	go func() {
-		for {
-			select {
-			case msg := <-p.outbox:
-				metrics.GetOrRegisterGauge("pss.outbox.len", nil).Update(int64(len(p.outbox)))
 
-				err := p.forward(msg.msg)
-				if err != nil {
-					log.Error(err.Error())
-					metrics.GetOrRegisterCounter("pss.forward.err", nil).Inc(1)
+	// In any case, if the message
+	// forwarding fails, the node should try to forward it to the next best peer, until the message is
+	// successfully forwarded to at least one peer.
+	go func() {
+		for slot := range p.outbox.process {
+			go func(slot int) {
+				msg := p.outbox.msg(slot)
+				sent := p.forwardFunc(msg.msg)
```

Comment:
so if forwardFunc returns an err then you could have the more common pattern

```golang
if err  := forwardFunc(o.msg(slot).msg); err != nil {
    log.Debug......
    reenqueue(slot)
    return
}
o.free(slot)

```

Reply:
FIxed in 1b91b66